### PR TITLE
[Docs] Add painless context details for bucket_selector

### PR DESCRIPTION
**Pull request by polyfractal**
Adds docs for `bucket_selector` context, including example and corresponding doc test

Similar to the `bucket_script` PR, I opted to only test this through the doc testing mechanism since the script itself is trivial.

Related to #34829

### CODE REVIEW

**Review by jdconrad — COMMENTED** [review]
@polyfractal Same as the bucket_script. Firstly, thank you! This looks good from the side to me.  I would like @debadair or another writer to take a look for the docs side.

File: docs/painless/painless-contexts.asciidoc [hunk 1]
```diff
@@ -46,6 +46,8 @@ specialized code may define new ways to use a Painless script.
                                     | {ref}/search-aggregations-metrics-scripted-metric-aggregation.html[Elasticsearch Documentation]
 | Bucket aggregation                | <<painless-bucket-agg-context, Painless Documentation>>
                                     | {ref}/search-aggregations-pipeline-bucket-script-aggregation.html[Elasticsearch Documentation]
+| Bucket Selector aggregation       | <<painless-bucket-selector-agg-context, Painless Documentation>>
+                                    | {ref}/search-aggregations-pipeline-bucket-selector-aggregation.html[Elasticsearch Documentation]
```
Comment (debadair) [inline docs/painless/painless-contexts.asciidoc]: Again, I'd lowercase "selector"

File: docs/painless/painless-contexts/painless-bucket-selector-agg-context.asciidoc [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+To run this example, first follow the steps in <<painless-context-examples, context examples>>.
+
+The painless context in a `bucket_selector` aggregation provides a `params` map.  This map contains both
+user-specified custom values, as well as the values from other aggregations specified in the `buckets_path`
+property.
+
+For our example, we are going to find the `max` of each bucket, add in a user-defined "base" value, then keep all
+the buckets that are greater than `10`.
+
```
Comment (debadair) [inline docs/painless/painless-contexts/painless-bucket-selector-agg-context.asciidoc]: Same as for bucket script:

This example finds the `max` of each bucket, adds a user-specified base_cost, and retains all of the buckets that are greater than `10`.

File: docs/painless/painless-contexts/painless-bucket-selector-agg-context.asciidoc [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+Unlike some other aggregation contexts, the `bucket_selector` context must return a boolean `true`/`false`.
+
```
Comment (debadair) [inline docs/painless/painless-contexts/painless-bucket-selector-agg-context.asciidoc]: I'm not a fan of using slashes to show options, I'd probably just use or,  `true` or `false`.

File: docs/painless/painless-contexts/painless-bucket-selector-agg-context.asciidoc [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+[source,Painless]
+--------------------------------------------------
+params.max + params.base_cost > 10
+--------------------------------------------------
+
+Note how all the values are being pulled from the `params` map. Also note how the script is in the form of an expression,
+which returns `true`/`false`.  In context, the aggregation looks like this:
+
```
Comment (debadair) [inline docs/painless/painless-contexts/painless-bucket-selector-agg-context.asciidoc]: Same as for bucket script:

Note that the values are extracted from the params map. The script is in the form of an expression that returns `true` or `false`.

File: docs/painless/painless-contexts/painless-bucket-selector-agg-context.asciidoc [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[setup:seats]
+<1> the `buckets_path` points to the max aggregations (`max_cost`) and adds `max` variables
+to the `params` map
+<2> the user-specified `base_cost` is added in the script's `params` section, which is also added to the
+`params` variable for use in the script.
```
Comment (debadair) [inline docs/painless/painless-contexts/painless-bucket-selector-agg-context.asciidoc]: Same as for buckets script--I'd init cap, consistently use `params` map, and drop "for use in the script".

**Review by debadair — CHANGES_REQUESTED** [review]
Suggested some minor edits.

**Comment by polyfractal** [comment]
Tidied up, and also re-arranged sentence order in one place.  :)

**Review by debadair — APPROVED** [review]

**Comment by polyfractal** [comment]
Jenkins, test this please.

**Pull request merged**